src/sap/fhir/model/r4/FHIRUtils.js

```diff
@@ -605,6 +606,52 @@ sap.ui.define([
 		return oBinding && oBinding.getMetadata().getName() === "sap.fhir.model.r4.FHIRPropertyBinding";
 	};
 
+	/**
+	 * Generate FullUrl based on the type and id
+	 *
+	 * @param {sap.fhir.model.r4.lib.FHIRBundleEntryFullUrlType} sFullUrlType fullUrlType
+	 * @param {string} sResourceServerPath The original resource path e.g. /Patient/123
+	 * @param {string} sResourceId The id of the the FHIR resource e.g. 123
+	 * @param {string} sServiceUrl The root URL of the FHIR server to request data e.g. http://example.com/fhir
+	 * @returns {string} sFullUrl
+	 * @protected
+	 * @since 1.0.0
+	 */
+	FHIRUtils.generateFullUrl = function(sFullUrlType,sResourceServerPath,sResourceId,sServiceUrl) {
+		var sFullUrl;
+		if (sFullUrlType){
+			switch (sFullUrlType) {
+				case FHIRBundleEntryFullUrlType.uuid:
+					if (sResourceId && this.resourceIdIsOfTypeUUID(sResourceId)){
```

Comment:
Thats dulicate code. I would prefer
sFullUrl = "urn:" + FHIRBundleEntryFullUrlType.uuid + ":";
if (sResourceId && this.resourceIdIsOfTypeUUID(sResourceId)){
sFullUrl += sResourceId;
} else {
sFullUrl += this.uuidv4(); 
}
``

Reply:
done